angular-frontend/src/service/get-api.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { Attendance } from 'src/Model/Attendance';
+import { $ } from 'protractor';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GetApiService {
+  constructor(private http: HttpClient) {}
+
+  headerDict = {
+    apiKey: 'yNvknZnTsO',
+  };
+
+  // api call function
+  GetAll(): Observable<Attendance[]> {
+    const requestOptions = { headers: new HttpHeaders(this.headerDict) };
+
+    return this.http.get<Attendance[]>(
+      'https://localhost:44353/api/Attendance',
+      requestOptions
+    );
+  }
+
+  GetProduct(Product: string): Observable<Attendance[]> {
+    const requestOptions = { headers: new HttpHeaders(this.headerDict) };
```

Comment:
Having this line at the start of each function is repetitive, and repetition is the enemy in code. You could move this to your constructor or init function, and store the `requestOptions` in a private class variable